core/src/replay_stage.rs

```diff
@@ -3089,15 +3090,48 @@ impl ReplayStage {
                     (bank.slot(), bank.hash()),
                     Some((bank.parent_slot(), bank.parent_hash())),
                 );
+
                 bank_progress.fork_stats.bank_hash = Some(bank.hash());
-                let bank_frozen_state = BankFrozenState::new_from_state(
+                let mut bank_frozen_state = BankFrozenState::new_from_state(
                     bank.slot(),
                     bank.hash(),
                     duplicate_slots_tracker,
                     duplicate_confirmed_slots,
                     heaviest_subtree_fork_choice,
                     epoch_slots_frozen_slots,
                 );
+
+                if bank.collector_id() != my_pubkey {
+                    // If the block does not have at least DATA_SHREDS_PER_FEC_BLOCK shreds in the last FEC set,
+                    // process it like a duplicate, which allows us to continue replaying the fork but not vote on it.
```

Comment:
Do we want to continue replaying this fork if it's clearly the wrong fork/doesn't meet the protocol requirement of 32 data shreds? Seems like it would be better to mark it as dead and let the duplicate logic handle getting another version if it's available.

Reply:
I see no benefit to marking it as dead instead of duplicate. Both cases will be handled by both the current and future duplicate confirmed resolution mechanisms.

Keeping it as a duplicate is preferable so we can continue replaying votes on the fork.

Reply:
talked more offline - will update this to mark as dead.
This is more inline with similar failures (poh failures), and we should not waste resources replaying any descendants. 
There is also no "duplicate proof" we can send in this scenario, so dead makes more sense.

Reply:
For future reference, maybe add the reasoning as a comment in the code why the block is marked as "dead" and not "duplicate"

Reply:
also, isn't there a concern that the block is rooted by the cluster anyways and so my node has to oblige? (which I am guessing if the block is marked as "dead" then it's not possible).
For example if the firedancer client does not implement this spec precisely, or we hit an edge case (or bug) with blockstore where the meta-data to confirm >=32 data-shreds isn't populated so the check fails only because the blockstore queries fail?

Reply:
Both "dead" and "duplicate" blocks are resolved similarly if >52% of the cluster votes on it. The differences are that:
* Dead blocks **always** require a dump & repair cycle in order to continue
* Duplicate blocks can be continued (marked un-duplicate) without dump & repair if we have the bank hash that matches what the cluster has voted on.